.github/workflows/smoke-checks.yml

```diff
@@ -232,52 +120,11 @@ jobs:
         path: /usr/local/lib/mint
         key: ${{ runner.os }}-mint-${{ hashFiles('./Mintfile') }}
         restore-keys: ${{ runner.os }}-mint-
-    - name: Run bootstrap.sh
-      run: ./bootstrap.sh
-    - name: Build Sample
+    - uses: ./.github/actions/bootstrap
+    - name: Build Sample App
       run: bundle exec fastlane build_sample
-    - uses: 8398a7/action-slack@v3
-      with:
-        status: ${{ job.status }}
-        text: "You shall not pass!"
-        fields: message,commit,author,action,eventName,workflow,job,took
-      env:
-        SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
-        MATRIX_CONTEXT: ${{ toJson(matrix) }}
-      if: ${{ github.event_name == 'push' && failure() }}
-
-  build-sample-ios12:
-    name: Build Sample (iOS12.4)
-    runs-on: macos-latest
-    steps:
-    - uses: actions/checkout@v1
-    - name: Set build image var
-      run: echo "ImageVersion=$ImageVersion"  >> $GITHUB_ENV
-    - name: Cache RubyGems
-      uses: actions/cache@v2
-      id: rubygem-cache
-      with:
-        path: vendor/bundle
-        key: ${{ runner.os }}-${{ env.ImageVersion }}-gem-${{ hashFiles('**/Gemfile.lock') }}
-        restore-keys: ${{ runner.os }}-${{ env.ImageVersion }}-gem-
-    - name: Cache Mint
-      uses: actions/cache@v2
-      id: mint-cache
-      with:
-        path: /usr/local/lib/mint
-        key: ${{ runner.os }}-mint-${{ hashFiles('./Mintfile') }}
-        restore-keys: ${{ runner.os }}-mint-
-    - name: Run bootstrap.sh
-      run: ./bootstrap.sh
-    - name: Prepare iOS 12 simulator
-      run: |
-        sudo mkdir -p /Library/Developer/CoreSimulator/Profiles/Runtimes
-        sudo ln -s /Applications/Xcode_10.3.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/CoreSimulator/Profiles/Runtimes/iOS.simruntime /Library/Developer/CoreSimulator/Profiles/Runtimes/iOS\ 12.4.simruntime
-        xcrun simctl list runtimes
-        xcrun simctl create custom-test-device "iPhone 7" "com.apple.CoreSimulator.SimRuntime.iOS-12-4"
-        xcrun simctl list devices 12.4
-    - name: Build Sample
-      run: bundle exec fastlane build_sample device:"iPhone 7 (12.4)"
+    - name: Build Demo App
+      run: bundle exec fastlane build_demo
```

Comment:
Do we want `Build Docs Snippets` as a part of smoke checks? Ideally, we'd keep the number of jobs <= 5

Reply:
This one is pretty fast to run and it is useful. I've already broken the docs snippets, and it was very easy to fix them because I had context on what broke. It would take much more time for someone else to fix it. So all the things that a PR can break, we should make sure it gets fixed there, so someone else won't be losing time before a release or something like that